src/coins.cpp

```diff
@@ -1102,3 +1171,136 @@ CCoinsModifier::~CCoinsModifier()
         cache.cachedCoinsUsage += it->second.coins.DynamicMemoryUsage();
     }
 }
+
+void SubtreeCache::clear() {
+    initialized = false;
+    parentLatestSubtree = std::nullopt;
+    newSubtrees.clear();
+}
+
+std::optional<libzcash::LatestSubtree> SubtreeCache::GetLatestSubtree(CCoinsView *parentView) {
```

Comment:
It seems odd to pass the parent view into individual methods. That won't work unless it's the same parent view each time, right?

Reply:
Yes, that must be the same parent view every time. There is no situation where it makes sense that the view would end up in a different state -- the rest of the `CCoinsViewCache` abstractions would break if it was misused in this way.